src/core/server/saved_objects/service/lib/filter_utils.ts

```diff
@@ -144,47 +130,55 @@ export const validateFilterKueryNode = (
 
 const getType = (key: string) => (key.includes('.') ? key.split('.')[0] : null);
 
-export const isSavedObjectAttr = (
-  key: string,
-  typeIndexPatterns: SavedObjectsIndexPatternField[]
-) => {
-  const splitKey = key.split('.');
-  if (splitKey.length === 1 && typeIndexPatterns.some(tip => tip.name === splitKey[0])) {
+/**
+ * Is this filter key referring to a a top-level SavedObject attribute such as
+ * `updated_at` or `references`.
+ *
+ * @param key
+ * @param indexMapping
+ */
+export const isSavedObjectAttr = (key: string, indexMapping: IndexMapping) => {
+  const keySplit = key.split('.');
+  if (keySplit.length === 1 && fieldDefined(indexMapping, keySplit[0])) {
     return true;
-  } else if (splitKey.length > 1 && typeIndexPatterns.some(tip => tip.name === splitKey[1])) {
+  } else if (keySplit.length === 2 && fieldDefined(indexMapping, keySplit[1])) {
     return true;
+  } else {
+    return false;
   }
-  return false;
 };
 
 export const hasFilterKeyError = (
   key: string,
   types: string[],
-  typeIndexPatterns: SavedObjectsIndexPatternField[]
+  indexMapping: IndexMapping
 ): string | null => {
   if (!key.includes('.')) {
     return `This key '${key}' need to be wrapped by a saved object type like ${types.join()}`;
   } else if (key.includes('.')) {
     const keySplit = key.split('.');
+
     if (keySplit.length <= 1 || !types.includes(keySplit[0])) {
       return `This type ${keySplit[0]} is not allowed`;
     }
     if (
-      (keySplit.length === 2 && typeIndexPatterns.some(tip => tip.name === key)) ||
-      (keySplit.length > 2 && types.includes(keySplit[0]) && keySplit[1] !== 'attributes')
```

Comment:
`types.includes(keySplit[0])` we wanted to make sure that the type who was used in the filter query was valid